source/framework/core/src/TRestMesh.cxx

```diff
@@ -420,9 +420,22 @@ Int_t TRestMesh::GetGroupId(Double_t x, Double_t y, Double_t z) {
 /// \brief Returns the group id using the position inside the nodes vector.
 ///
 Int_t TRestMesh::GetGroupId(Int_t index) {
-    if (index > (int)nodeGroupID.size()) return GROUP_NOT_FOUND;
+    if (index > (int)fNodeGroupID.size()) return GROUP_NOT_FOUND;
 
-    return nodeGroupID[index];
+    return fNodeGroupID[index];
+}
+
+///////////////////////////////////////////////
+/// \brief It returns the total energy of all nodes corresponding to the group id given by argument
+///
+Double_t TRestMesh::GetGroupIdEnergy(Int_t index) {
+    if (index > (int)fNodeGroupID.size()) return 0.0;
```

Comment:
This line is needed? Assuming GetNumberOfNodes() == 0, the lines below will just return 0...